### PR TITLE
chore: add initial draft of the IDM spec

### LINKED ISSUE #14: IDM Spec paper
## Description

Have a draft for the IDM specification so that it can be implemented by anyone.

## Acceptance Criteria

- [ ] Have a draft document specifying the IDM Wallet, IDM Client and IDM Bridge interfaces as well as any protocols (e.g.: replication)
- [ ] Ensure that the document is updated in the RWOT repo

### PR DESCRIPTION
**Pull request by satazor**
This commit has a very initial draft of the IDM specification. Note that it's a living document and will be updated throughout the project.

Closes #14

### CODE REVIEW

File: docs/idm-spec.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+.locker.unlock(LockerType type, (Any challenge) => Any solution): Promise<PrivateKey>
+.locker.lock()
+.locker.isLocked(): Boolean
+.locker.onLockedChange((Boolean locked) => {}): Function (to remove the listener)
+.locker.secret.getPublicKey(): Promise<PublicKey>
+.locker.secret.getPrivateKey(): Promise<PrivateKey>
+.locker.idle.setMaxTime(Number idleTime)
+.locker.idle.getMaxTime(): Number
+.locker.idle.getRemainingTime(): Number
+.locker.idle.reset():
```
Comment (paulobmarcos) [inline docs/idm-spec.md]: Extra `:` at the end

File: docs/idm-spec.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+.locker.locks.list(): Promise<Array<LockerType>>
+.locker.locks.set(LockerType type, Any solutions): Promise
+.locker.locks.drySet(LockerType type, Any solutions): Promise
+.locker.locks.unset(LockerType type): Promise
+```
+
+### .storage
+
+```js
+.storage.get(String<String>|String key): Promise<Object<String,Any>>
```
Comment (paulobmarcos) [inline docs/idm-spec.md]: Is `String<String>` meant to be `Array<String>`?

Comment (satazor) [reply docs/idm-spec.md]: Fixed

File: docs/idm-spec.md (hunk 1 above)
Comment (satazor) [reply docs/idm-spec.md]: Fixed

**Comment by satazor** [comment]
@paulobmarcos all fixed. Pleas hit squash and merge as I added a commit without amending (did on my phone).

**Pull request merged**